contractcourt/htlc_outgoing_contest_resolver_test.go

```diff
@@ -0,0 +1,158 @@
+package contractcourt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet"
+)
+
+const (
+	outgoingContestHtlcExpiry = 110
+)
+
+// TestHtlcOutgoingResolverTimeout tests resolution of an offered htlc that
+// timed out.
+func TestHtlcOutgoingResolverTimeout(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	ctx := newOutgoingResolverTestContext(t)
+	ctx.resolve()
+	ctx.notifyEpoch(outgoingContestHtlcExpiry - 1)
+	ctx.waitForResult(true)
+}
+
+// TestHtlcOutgoingResolverRemoteClaim tests resolution of an offered htlc that
+// is claimed by the remote party.
+func TestHtlcOutgoingResolverRemoteClaim(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	ctx := newOutgoingResolverTestContext(t)
+	ctx.resolve()
+	preimage := lntypes.Preimage{}
+	ctx.notifier.spendChan <- &chainntnfs.SpendDetail{
```

Comment:
same, needs comments, otherwise very hard to know what's going on

Reply:
yes fair point. added